### PR TITLE
Add collection contract

### PR DESCRIPTION
**Pull request by hyperspacebunny**

### CODE REVIEW

File: contracts/collection/NFTCollectionContract.sol [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./templates/NFTCollection.sol";
+
+contract NFTCollectionContract is NFTCollection {
+    constructor(
```
Comment (kristjanpeterson1) [inline contracts/collection/NFTCollectionContract.sol:7]: why is this necessary?

File: contracts/collection/templates/NFTCollection.sol [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+    /// The number of currently minted tokens
+    /// @dev Managed by the contract
+    uint256 public totalSupply;
+
+    /***************************
+     * Contract initialization *
+     ***************************/
+
+    constructor() ERC721("", "") {
```
Comment (kristjanpeterson1) [inline contracts/collection/templates/NFTCollection.sol:103]: btw dont we need name and symbol in ERC721 constructor? :O

File: contracts/collection/templates/NFTCollection.sol [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+    function presaleActive() public view returns (bool) {
+        return block.timestamp > _runtimeConfig.presaleMintStart;
+    }
+
+    /// Get the number of tokens still available for minting
+    function availableSupply() public view returns (uint256) {
+        return
+            _deploymentConfig.maxSupply -
+            totalSupply -
+            _deploymentConfig.reservedSupply;
```
Comment (kristjanpeterson1) [inline contracts/collection/templates/NFTCollection.sol:169]: what if reserversupply bigger than max supply?

Comment (kristjanpeterson1) [reply contracts/collection/templates/NFTCollection.sol:169]: ah i see _validateDeploymentConfig takes care of that

**Review by kristjanpeterson1 — APPROVED** [review]

File: contracts/collection/templates/NFTCollection.sol (hunk 2 above)
Comment (hyperspacebunny) [reply contracts/collection/templates/NFTCollection.sol:103]: Not really, we override `name()` and `symbol()` later in the `Internals` section so the contract still has them implemented properly

**Review by KasparPeterson — APPROVED** [review]

File: contracts/lib/Base64.sol [hunk 4]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: MIT
-pragma solidity ^0.8.4;
+pragma solidity ^0.8.0;
```
[the remark this answers is not in the document]
Comment (hyperspacebunny) [reply contracts/lib/Base64.sol:2]: No particular reason, this got changed while I was trying to fix MythX and it's just what the other collection contract files were using
